libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -151,19 +155,42 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 
 string IRGenerator::constructorCode(ContractDefinition const& _contract)
 {
-	// TODO initialize state variables in base to derived order.
-	// TODO base constructors
-	// TODO callValueCheck if there is no constructor.
+	using boost::adaptors::reverse;
+
 	if (FunctionDefinition const* constructor = _contract.constructor())
 	{
 		string out;
 		if (!constructor->isPayable())
 			out = callValueCheck();
+
 		solUnimplementedAssert(constructor->parameters().empty(), "");
-		return move(out) + m_context.functionName(*constructor) + "()\n";
+		solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
+
+		// Initialization of state variables in base-to-derived order.
+		for (ContractDefinition const* contract : reverse(_contract.annotation().linearizedBaseContracts))
+			for (VariableDeclaration const* variable : contract->stateVariables())
+				out += initializeStateVariable(*variable);
+
+		// TODO base constructors
+
+		out += m_context.functionName(*constructor) + "()\n";
+
+		return out;
 	}
+	else
+		return callValueCheck(); // callValueCheck if there is no constructor.
+}
+
+string IRGenerator::initializeStateVariable(VariableDeclaration const& _varDecl)
+{
+	// something like: SSTORE(storageSlot, expressionValue)
+
+	IRGeneratorForStatements generator{ m_context, m_utils };
+	_varDecl.value()->accept(generator);
+	string const rightHandSide = generator.code();
 
-	return {};
+	IRStorageItem lvalue{m_context, _varDecl};
+	return lvalue.storeValue(rightHandSide, *_varDecl.type());
```

Comment:
I think you need something like
```suggestion
	return lvalue.storeValue(m_context.variable(_varDecl.value()), *_varDecl.type());
```

Reply:
And also it shouldn't be `_varDecl.type()` but `_varDecl.vaule().annotation().type.`